src/Jaeger.Core/Tracer.cs

```diff
@@ -165,7 +181,7 @@ public sealed class Builder
             private ISampler _sampler;
             private IReporter _reporter;
             private PropagationRegistry _registry;
-            private IMetrics _metrics;
+            private IMetrics _metrics = new MetricsImpl(NoopMetricsFactory.Instance);
```

Comment:
Don't define it here, since `Build` has the fallback code which we need anyway in case someone calls `WithMetrics(null)`.

Reply:
I don't like this either. This was done for parity with Java. I want to create a separate PR that fixes this behavior for all builders afterwards (for Java and C#).

Reply:
It‘s ok to keep it for the PR but this is also unnecessary code in Java since it‘s redundant and defies the purpose of the setter since it can’t become null if not set explicitly.